documentation/developer_setup.md

```diff
@@ -0,0 +1,126 @@
+	resp, err := t.client.Create(ctx, TableProperties{&name}, new(TableCreateOptions), new(QueryOptions))
+	if err == nil {
+		tableResp := resp.(TableCreateResponse)
+		return tableResp, nil
+	}
+	return TableCreateResponse{}, err
+}
+```
+
+All methods that perform I/O of any kind, sleep, or perform a significant amount of CPU-bound work must have the first parameter be of type [`context.Context`][golang_context] which allows the customer to carry a deadline, cancellation signal, and other values across API boundaries. The remaining parameters should be parameters specific to that method. The return types for methods should be first a "Response" object and second an `error` object.
+
+## Write Tests
+
+Testing is built into the Go toolchain as well with the `testing` library. The testing infrastructure located in the `sdk/internal` directory takes care of generating recordings, establishing the mode a test is being run in (options are "recording", "playback", "live-no-playback"), and reading environment variables.
+
+A simple test for `aztables` is shown below:
+```golang
+
+import (
+	"os"
+
+	"github.com/testify/assert"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
+)
+
+const (
+	accountName := os.GetEnv("TABLES_PRIMARY_ACCOUNT_NAME")
+	accountKey := os.GetEnv("TABLES_PRIMARY_ACCOUNT_KEY")
+	mode := recording.Recording
+)
+
+// Test creating a single table
+func TestCreateTable(t *testing.T) {
+	client := NewTableClient(accountName, accountKey, "tableName")
+	resp, err := client.Create()
+	assert.Nil(t, err)
+	assert.Equal(t, respo.TableResponse.TableName, "tableName")
+}
+```
+
+The first part of the test above is for getting the secrets needed for authentication from your environment, the current practice is to store your test secrets in environment variables.
+
+The rest of the snippet shows a test that creates a single table and asserts that the response from the service has the same table name as the supplied parameter. Every test in Go has to have exactly one parameter, the `t *testing.T` object, and it must begin with `Test`. After making a service call or creating an object you can make assertions on that object by using the external `testify/assert` library. In the example above, we assert that the error returned is `nil`, meaning the call was successful and then we assert that the response object has the same table name as supplied.
+
+You can also use the `testify/require` library instead of `testify/assert` if you want your test to fail as soon as you have an unexpected result.
+
+Check out the docs for more information about the methods available in the [`assert`](https://pkg.go.dev/github.com/stretchr/testify/assert) and [`require`](https://pkg.go.dev/github.com/stretchr/testify/require) libraries.
+
+
+<!-- LINKS -->
+[workspace_setup]: https://www.digitalocean.com/community/tutorials/how-to-install-go-and-set-up-a-local-programming-environment-on-windows-10
```

Comment:
This tutorial is probably more than is needed. 

For Windows, there's an actual MSI that you can use to install and that's about it. No chocolately needed (and certainly no nano!)